api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1826,12 +1826,10 @@ type AzurePlatformSpec struct {
 // OpenStackPlatformSpec specifies configuration for clusters running on OpenStack.
 type OpenStackPlatformSpec struct {
 
-	// Name of the secret that contains OpenStack credentials needed for creating and managing cloud
-	// infrastructure resources.
-	//
+	// IdentityRef is a reference to a secret holding OpenStack credentials
+	// to be used when reconciling the hosted cluster.
 	// +kubebuilder:validation:Required
-	// +required
-	CloudsYamlSecret corev1.LocalObjectReference `json:"cloudsYamlSecret"`
+	IdentityRef OpenStackIdentityReference `json:"identityRef"`
 
 	// Name of the secret that contains the CA cert needed to connect to the OpenStack cloud.
 	// +optional
```

Comment:
what to do with this one? I haven't figured whether we want to keep it.

Reply:
I guess we could get rid of it by checking in the secret whether we have a CA cert.